Sources/FoundationExtensions/AsyncExtensions.swift

```diff
@@ -139,7 +139,7 @@ internal enum Async {
         repeat {
             retries += 1
             let (shouldRetry, result) = await value()
-            if shouldRetry {
+            if !shouldRetry {
```

Comment:
Nit: change this to `shouldRetry` and move this to the else and viceversa. Easier to read than the reverse logic.